server/commands/write_test.go

```diff
@@ -89,3 +91,68 @@ func TestValidateWriteTuples(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateWriteTuples(t *testing.T) {
+	type test struct {
+		name          string
+		deletes       []*openfgapb.TupleKey
+		writes        []*openfgapb.TupleKey
+		expectedError error
+	}
+
+	tracer := telemetry.NewNoopTracer()
+	logger := logger.NewNoopLogger()
+	dbCounter := utils.NewDBCallCounter()
+
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	maxTuplesInWriteOp := 10
+	mockDatastore := mockstorage.NewMockOpenFGADatastore(mockController)
+	mockDatastore.EXPECT().MaxTuplesInWriteOperation().AnyTimes().Return(maxTuplesInWriteOp)
+
+	badItem := &openfgapb.TupleKey{
+		Object:   fmt.Sprintf("%s:1", testutils.CreateRandomString(459)),
+		Relation: testutils.CreateRandomString(50),
+		User:     "",
+	}
+
+	cmd := NewWriteCommand(mockDatastore, tracer, logger)
+
+	tests := []test{
+		{
+			name:          "nil for deletes and writes",
+			deletes:       nil,
+			writes:        nil,
+			expectedError: serverErrors.InvalidWriteInput,
+		},
+		{
+			name:          "validation on tuples against write",
+			deletes:       []*openfgapb.TupleKey{},
+			writes:        []*openfgapb.TupleKey{badItem},
+			expectedError: serverErrors.InvalidTuple("missing user", badItem),
+		},
+		{
+			name:          "validation on tuples against delete",
```

Comment:
Can we rename this to a more appropriate name such as "delete failure with invalid user".